### ISSUE TITLE
Discussion: Mode() to display null values as mode

### ISSUE THREAD

**Issue by sauren-khosla** [influxdata/flux#1617]
Starting a discussion:
Should `mode()` display `null` values if `null` is also a mode?

Some things to consider: 
1. Including null values notifies the user that there is possibly an irregularity in their dataset
2. null == null returns false, so multiple null values may not actually be the same null

**Comment by aanthony1243** [comment]
i was convinced by @nathanielc  and his thought that in the database-world,   `null == null` is false.  That is, you can't know what you don't know.  In this context, the mathematical definition of `mode` would necessarily exclude null.

**Comment by nathanielc** [comment]
In light of the above we should probably state that like other aggregate transformations that `mode` has an implicit filter on `nulls`.

**Comment by wolffcm** [comment]
This makes sense.

However, if the column contains *only* null values, then I would be inclined to make `mode()` return `null` just for that case.

**Comment by jlapacik** [comment]
I agree with the predominant sentiment. Aggregates should almost always filter out nulls .
